components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -9158,6 +9158,55 @@ public static JSONObject executeQueryOnStreamProcessor(String appName, String qu
 
     }
 
+    /**
+     * Implemented to get the API usage count for monetization.
+     *
+     * @param from : the start timestamp of the query.
+     * @param to   : the end timestamp of the query.
+     * @return JSON Object.
+     */
+    public static JSONObject getUsageCountForMonetization(long from, long to)
+            throws APIManagementException {
+
+        JSONObject jsonObject = null;
+        String granularity = null;
+        APIManagerConfiguration configuration = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
+                .getAPIManagerConfiguration();
+        granularity = configuration.getFirstProperty(
+                APIConstants.Monetization.USAGE_PUBLISHER_GRANULARITY);
+        if (StringUtils.isEmpty(granularity)) {
+            //set the default granularity to days, if it is not set in configuration
+            granularity = APIConstants.Monetization.USAGE_PUBLISH_DEFAULT_GRANULARITY;
+        }
+        StringBuilder query = new StringBuilder(
+                "from " + APIConstants.Monetization.MONETIZATION_USAGE_RECORD_AGG
+                        + " within " + from
+                        + "L, " + to + "L per '" + granularity
+                        + "' select "
+                        + APIConstants.Analytics.API_NAME + ", "
+                        + APIConstants.Analytics.API_VERSION + ", "
+                        + APIConstants.Analytics.API_CREATOR + ", "
+                        + APIConstants.Analytics.API_CREATOR_TENANT_DOMAIN + ", "
+                        + APIConstants.Analytics.APPLICATION_ID + ", "
+                        + "sum (requestCount) as requestCount "
+                        + "group by "
+                        + APIConstants.Analytics.API_NAME + ", "
+                        + APIConstants.Analytics.API_VERSION + ", "
+                        + APIConstants.Analytics.API_CREATOR + ", "
+                        + APIConstants.Analytics.API_CREATOR_TENANT_DOMAIN + ", "
+                        + APIConstants.Analytics.APPLICATION_ID
+        );
+        try {
+            jsonObject = APIUtil.executeQueryOnStreamProcessor(
+                    APIConstants.Monetization.MONETIZATION_USAGE_RECORD_APP,
+                    query.toString());
+        } catch (APIManagementException ex) {
+            String msg = "Unable to Retrieve monetization usage records";
+            handleException(msg, ex);
+        }
+        return jsonObject;
```

Comment:
JsonObject can be null

Reply:
fixed in a129c3ec166bfd118842b18f7cbadb695d40f206